### ISSUE TITLE
.zip gTLD Blocking

### ISSUE THREAD

**Issue by 9ary** [greyhat-academy/lists.d#29]
Hello there!

First of all, I just want to say that I come in peace. Even though I don't currently have any use for these lists myself, I find compilation efforts like this one highly valuable. I just came across f341ed1 the other day and wanted to provide some input. The following post is my subjective opinion, and while I would be happy if you find it valuable, it's entirely up to you to decide what to make of it.
As a disclaimer, I own a .zip domain. Despite that, I don't feel entitled to being unblocked (especially since it's mainly for internal use). The above still applies.

I think blocking an entire TLD should be a last resort measure.

A good example that would justify it is the TLDs that Freenom (used to) make available for free (https://krebsonsecurity.com/2023/03/sued-by-meta-freenom-halts-domain-registrations/), which were not only used by abusers, but the registrar's handling of abuse reports was worse than inadequate.

I think we can agree that Google should be minimally trusted, but in my opinion, it's still too early to decide that the .zip TLD is only used for nefarious purposes. I would argue that the minimum cost of entry (15USD) actually makes it somewhat of a deterrent in a lot of cases. I'm not aware of Google having a similar history of harboring abusers. I think as long as they handle abuse reports properly the threat should hopefully be mitigated.

I understand the concerns brought up by the security community regarding the high potential for abuse of a TLD that is named after a popular file name extension, and I take them seriously. However I think it's also important to keep in mind that there is really nothing new in the space of misleading URLs. For example, it's been possible to craft homograph domains for a long time, and browser mitigations can only help so much (but *should* be constantly refined to help as much as possible). The .zip TLD adds a new way to confuse users, but within the existing context of similar attacks I would say that the threat is fairly diluted.

Some criticism of the new TLD I came across:

- https://medium.com/@bobbyrsec/the-dangers-of-googles-zip-tld-5e1e675e59a5
  - the homograph attack described in this post was reported to Chromium in 2016, and closed as wontfix, but it looks like they might be reconsidering: https://bugs.chromium.org/p/chromium/issues/detail?id=584644

On the other hand, I am aware of legitimate uses of the TLD. Samples from my browsing history:
- single pages calling out its potential for abuse:
  - https://financialstatement.zip
  - https://document.zip
  - https://master.zip
- jokes:
  - rickrolls:
    - https://attachment.zip
    - https://chatgpt3.zip
  - https://archlinux.zip (redirects to https://echodolls.com/)
  - misc sites that return a specific HTTP status code:
    - https://teapot.zip (418 I'm A Teapot)
    - https://yourdad.zip (410 Gone)
    - https://yourmom.zip (413 Request Entity Too Large)
    - https://yourmum.zip (ditto)
    - https://yourwife.zip (randomized responses, ~but it seems like it's down~)
- personal sites:
  - https://kwak.zip (mine, the landing page is just a spinning duck at the moment, but one public-facing use I intend for it is email)
  - https://friendshipcastle.zip

Overall I'm not sure it's exactly a good idea to block the entire TLD (yet). On the one hand, one wouldn't lose much currently by doing so (only one of the sites listed above currently has meaningful content), but on the other hand there are legitimate users, and it's safe to say there will be more in the future.

**Comment by kkarhan** [comment]
Hi @9ary,
first of all, sorry for my late reply, as I was quite busy.

You are kinda right in terms of criticism and thus I've out-commented .zip from the list.
Needless to say the lack of bona-fide domains - unlike with Freenom-maintained ccTLDs - is a major issue.

I do consider making a supplementary blocklist similar to [milintel.ipv4.block.list.tsv](https://github.com/greyhat-academy/lists.d/blob/main/milintel.ipv4.block.list.tsv) instead to encompass them.

From my own personal and professional experience, the core issues are hosters that don't do proper abuse-handling at all like GMail, YahooMail, Outlook.com / Azure Hosted Exchange and GoDaddy that literally ignore abure report emails!

**Comment by kkarhan** [comment]
@9ary you're free to make a PR for [drop.domains.block.list.tsv](https://github.com/greyhat-academy/lists.d/blob/main/drop.domains.block.list.tsv) with known malicious or abusive domains under .zip or any other TLD...

Please let me know if that solves the issue for you.
I also thank you for being forthcoming about your relation with said ccTLD, even if that won't have any impact on the decision.

**Comment by 9ary** [comment]
Thanks for replying! I was mainly looking to open discussion after finding this repo from a link on the fediverse. I think you've addressed my concerns.

> From my own personal and professional experience, the core issues are hosters that don't do proper abuse-handling at all like GMail, YahooMail, Outlook.com / Azure Hosted Exchange and GoDaddy that literally ignore abure report emails!

Yes, this is exactly what I was thinking of when I quoted Freenom. It's quite unfortunate that it took Meta suing them of all things to make them stop.

I'll also mention a more recent development that is related to this: Google Domains (the registrar) is being sold to Squarespace (https://support.google.com/domains/answer/13689670). As far as I can tell, Google remains the registry for the TLDs it currently owns, but going forward it looks like those domains are going to be resold by third-party registrars only (and .zip is already available on porkbun). This at least means that Google's extremely lacking support shouldn't be in the loop anymore (hopefully?).

> @9ary you're free to make a PR for [drop.domains.block.list.tsv](https://github.com/greyhat-academy/lists.d/blob/main/drop.domains.block.list.tsv) with known malicious or abusive domains under .zip or any other TLD...

I'll be sure to report any that I find over here.

Also, another post I've read since writing the OP, this time in defense of the new TLDs: https://textslashplain.com/2023/05/13/new-tlds-not-bad-actually/

**Comment by kkarhan** [comment]
@9ary thanks for your reply.

Whilst I do personally think that the entire domain namespace is sadly a huge clusterfuck, I do know that some places have literally gone so far to block .zip & .mov as gTLDs entirely.

That being said this issue would not exist if a basic level of tech-literacy was required to use a computer [similar to driving a motorized vehicle] and people as well as organizations would stop using insecure Govware like Windows as their OS.

**Comment by 9ary** [comment]
While I think the URL issue is pretty subtle and can confuse even seasoned users, I do agree that there is a huge problem with tech literacy, and it runs much deeper than people not understanding the tool because they just don't care. Computers are deliberately framed as magic black boxes in order to minimize their usefulness as a tool that serves, rather than exploit, the user.

Also noteworthy is that the past couple decades have shown that educating users on phishing is very ineffective. This is the whole reason why multi-factor authentication is being used, and more resistant forms of it are being developed.
As far as downloading malicious files, it's very difficult to protect anyone who's not expecting an attack. Antivirus software is of limited usefulness.

**Comment by kkarhan** [comment]
@9ary I've to fully agree on that, because Antivirus - espechally 3rd party solutions on Windows - are more value-removing than anything...

I'll close this issue then as it seems resolved.
Should there be need to reopen it, feel free to to do so anytime.
